### PR TITLE
Add a slash at the end of a requested URI

### LINKED ISSUE #896: Allow access without trailing slash for site-url/glotpress
Hello,

I just installed the plugin and was trying to access Glotpress and got a 404 until I added the trailing slash.

So this url - http://glotpress.eclipsedev.tk/glotpress
Loads the WP 404

But if I supply the trailing slash - http://glotpress.eclipsedev.tk/glotpress/
Then it redirects to - http://glotpress.eclipsedev.tk/glotpress/projects/

Can the redirect logic be updated to function when no tailing slash is there.

Thanks

### PR DESCRIPTION
**Pull request by amieiro**
## Problem

When you access the https://glotpress.test/glotpress URI (without a slash at the end of the URI), you get a 404. Instead, if you add the slash at the end of the URI https://glotpress.test/glotpress/, you are redirected to https://glotpress.test/glotpress/projects/

Keep in mind, I am using https://glotpress.test/ as my main WordPress address. Please, change all the URI used in this description to the main URI used in your local installation.

Fixes https://github.com/GlotPress/GlotPress/issues/896.

## Solution

This PR adds a slash at the end of the requested URI if it doesn't have it.

## Testing Instructions

Test both URI in your local installation (with and without the final slash) before and after applying this PR:
1. https://glotpress.test/glotpress
2. https://glotpress.test/glotpress/

Before applying the PR, you will get a 404 with the first URI. After applying the PR, you will be redirected to the GlotPress main page.

### CODE REVIEW

**Review by trymebytes — DISMISSED** [review]

**Comment by trymebytes** [comment]
Tested and works for me.

File: gp-includes/router.php [hunk 1]
```diff
@@ -181,6 +181,10 @@ public function route() {
 
 		$url_path = gp_url_path( gp_url_public_root() );
 
+		// Check if the last character of the text is not a slash, and add it if needed
+		if ( substr( $_SERVER['REQUEST_URI'], -1 ) !== '/' ) {
+			$_SERVER['REQUEST_URI'] .= '/';
```
Comment (akirk) [inline gp-includes/router.php]: If we need to modify `$_SERVER['REQUEST_URI']` then we'd use `trailingslashit()`

**Review by akirk — CHANGES_REQUESTED** [review]
I am skeptic of this approach. For routing [there is a function `request_uri()`](https://github.com/GlotPress/GlotPress/blob/eff0b084962da725e0c124eb99ecbd9e19686544/gp-includes/router.php#L32). I think we should be working with that instead of modifying a PHP superglobal.

File: gp-includes/router.php [hunk 2]
```diff
@@ -181,6 +181,10 @@ public function route() {
...
+		}
 		// If the request URL doesn't match our base URL, don't bother trying to match
 		if ( $url_path && ! gp_startswith( wp_unslash( $_SERVER['REQUEST_URI'] ), $url_path ) ) {
```
Comment (akirk) [inline gp-includes/router.php]: Can we also do this?
```suggestion
		if ( $url_path && ! gp_startswith( wp_unslash( trailingslashit( $_SERVER['REQUEST_URI'] ) ), $url_path ) ) {
```

File: gp-includes/router.php (hunk 1 above)
Comment (amieiro) [reply gp-includes/router.php]: Removed

**Review by akirk — APPROVED** [review]
Thank you!

**Pull request merged**